include/nfc/t2t/parser.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#ifndef NFC_T2T_PARSER_H_
+#define NFC_T2T_PARSER_H_
+
+#include <stdint.h>
+#include <nfc/t2t/tlv_block.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file
+ * @defgroup nfc_t2t_parser Type 2 Tag
+ * @{
+ * @brief NFC TYPE 2 TAG parser.
+ */
+
+/** @brief Descriptor for the internal bytes of a Type 2 Tag.
+ */
+struct nfc_t2t_serial_number {
+	/** Manufacturer ID (the most significant byte of the
+	 *  UID/serial number).
+	 */
+	u8_t manufacturer_id;
+
+	/** Bytes 5-4 of the tag UID. */
+	u16_t serial_number_part_1;
+
+	/** First block check character byte (XOR of the cascade tag byte,
+	 *  manufacturer ID byte, and the serial_number_part_1 bytes).
+	 */
+	u8_t check_byte_0;
+
+	/** Bytes 3-0 of the tag UID. */
+	u32_t serial_number_part_2;
+
+	/** Second block check character byte
+	 * (XOR of the serial_number_part_2 bytes).
+	 */
+	u8_t check_byte_1;
+
+	/** Tag internal bytes. */
+	u8_t internal;
+};
+
+/** @brief Descriptor for the Capability Container (CC) bytes of a Type 2 Tag.
+ */
+struct nfc_t2t_capability_container {
```

Comment:
I think you can use shortcuts here like nfc_t2t_cc

Reply:
Ok